iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ExecFormImpl.java

```diff
@@ -58,11 +59,21 @@ public SyntaxToken leftBracket() {
     return leftBracket;
   }
 
+  /**
+   * @deprecated To be removed once arguments() methods exist in all implementation and that literals() can be replaced everywhere.
+   * For now the method has been transformed to still provide the same data as before.
+   */
+  @Deprecated(forRemoval = true)
   @Override
   public List<SyntaxToken> literals() {
-    List<SyntaxToken> result = new ArrayList<>();
-    for (ExecFormLiteral element : literalsWithSeparators.elements()) {
-      result.add(element.value());
+    throw new UnsupportedOperationException("Method to be removed from LiteralList interface once SONARIAC-541 and SONARIAC-572 are done.");
+  }
+
+  @Override
+  public List<Argument> arguments() {
+    List<Argument> result = new ArrayList<>();
+    for (ExpandableStringLiteral element : expressionsWithSeparators.elements()) {
+      result.add(new ArgumentImpl(List.of(element)));
```

Comment:
The tree class should not take care of this. Maybe we can let the grammar provide the arguments in the separate list directly? WDYT? (However, we can do it in another PR)

Reply:
I agree on this, I put this in place in my last commit.